server/api/section/section.controller.js

```diff
@@ -61,6 +61,58 @@ function removeEntity(res) {
   };
 }
 
+function updateSection(req){
+  if(req.body.pendingStudent){
+    return updatePendingStudents(req);
+  }
+  else{
+    return saveSectionUpdates(req);
+  }
+}
+
+function updatePendingStudents(req){
+  return (section) =>{
+    var pendingStudent = req.body.pendingStudent;
+    if(section.pendingStudents.remove(pendingStudent)){
+      section.students.push(pendingStudent);
+      return section.saveAsync()
+        .spread((sect)=>{
+          return Section.populate(sect, {path:"pendingStudents"})
+        });
+    }
+  }
+}
+
+function checkSectionReq(req) {
+  // Check sections numbers
+  var sectionNumbers = req.body.sectionNumbers.map(Number);
+  if(sectionNumbers){
+    sectionNumbers.sort();
+    for (var i = 0; i < sectionNumbers.length - 1; i++) {
+      if(sectionNumbers[i] < 0){
+        throw "Section number must be greater than zero";
+      }
+      if (sectionNumbers[i + 1] == sectionNumbers[i]) {
+        throw "Duplicate section numbers found";
+      }
+    }
+  }
```

Comment:
:penguin: There isn't a utility method that does this duplicate checking in `lodash`?